packages/wdio-utils/tests/driver/utils.test.ts

```diff
@@ -30,30 +46,76 @@ vi.mock('node:fs', () => ({
             'SetupMetrics'
         ])
     }
-}))
 
-vi.mock('node:fs/promises', () => ({
-    default: {
+    return {
+        ...origFS,
+        ...mod,
+        default: {
+            ...origFS,
+            ...mod
+        }
+    }
+})
+
+vi.mock('node:fs/promises', async () => {
+
+    const origFS = await vi.importActual<typeof import('node:fs/promises')>('node:fs/promises')
+
+    const mod = {
         mkdir: vi.fn().mockResolvedValue({}),
         access: vi.fn().mockResolvedValue({})
     }
-}))
 
-vi.mock('node:child_process', () => ({
-    default: {
+    return {
+        ...origFS,
+        ...mod,
+        default: {
+            ...origFS,
+            ...mod,
+        }
+    }
+})
+
+vi.mock('node:child_process', async () => {
+
+    const origCP = await vi.importActual<typeof import('node:child_process')>('node:child_process')
```

Comment:
I honestly would not mix actual implementation an mock together. We should just enhance the mock file rather.